common/reviews/api/api-extractor.api.md

```diff
@@ -74,8 +78,9 @@ export class ExtractorConfig {
     readonly projectFolder: string;
     readonly projectFolderUrl: string | undefined;
     readonly publicTrimmedFilePath: string;
-    readonly reportFilePath: string;
-    readonly reportTempFilePath: string;
```

Comment:
Removing `reportFilePath` and `reportTempFilePath` would be a breaking change to the API. 

In the existing design, they return an empty string when the API report feature is disabled.  Maybe we could just change these settings to return the `"complete"` variant, only if that variant is enabled.  We could also mark them as `@deprecated`.

Reply:
Oh, I missed that this class was exported. Yeah, I'll revert the changes here.

Reply:
Question: are consumers expected to extend this class at all? Would it be safe to add legacy getter properties (possibly marked as deprecated)?

Also wondering if this class should be marked as `@sealed`.

Reply:
I went ahead and implemented the getter property route but let me know if you think that is insufficient. I'm also happy to mark the class as `@sealed` while I'm here, if we think that's the right call.

Reply:
>  are consumers expected to extend this class at all?

Consumers can't inherit `ExtractorConfig` because the constructor is private.

> Also wondering if this class should be marked as `@sealed`.

Sure

Reply:
> Would it be safe to add legacy getter properties (possibly marked as deprecated)?

Yes this is fine. Since it's a `class` we can use logic to defer computation.  (That is also a potential reason for `IExtractorConfigApiReport` to be a `class`, but I didn't bring this up since it seems probably immaterial.)

However according to our coding conventions, property getters should not normally throw exceptions.

Reply:
> Consumers can't inherit ExtractorConfig because the constructor is private.

Ah, missed that the constructor was private. Excellent.

> However according to our coding conventions, property getters should not normally throw exceptions.

None of what's in the getter implementations should throw. Can you elaborate?

Reply:
> None of what's in the getter implementations should throw. Can you elaborate?

The code's fine. ( This comment was like when GPT keeps blathering about stuff you didn't ask and you have to press the ⏹️ button. 😄 )